src/components/dialogs/load-modification-dialog.js

```diff
@@ -97,18 +105,55 @@ const LoadModificationDialog = ({
         loading: loadingEquipmentOptions,
     });
 
+    const id = useMemo(() => {
+        let id;
+        if (typeof loadId === 'object') {
+            if (loadId?.id !== '') {
+                id = loadId?.id;
+            }
+        } else {
+            if (loadId !== '') {
+                id = loadId;
+            }
+        }
+        return id ?? formValueEquipmentId?.id;
+    }, [loadId, formValueEquipmentId]);
+
+    const fetchGeneratorInformation = async (
+        studyUuid,
+        currentNodeUuid,
+        equipmentId
+    ) => {
+        const value = await fetchEquipmentInfos(
+            studyUuid,
+            currentNodeUuid,
+            'loads',
+            equipmentId,
+            false
+        );
+        if (value) {
+            setloadInfo(value);
+        }
+    };
+
+    useEffect(() => {
+        if (id) {
+            fetchGeneratorInformation(studyUuid, currentNodeUuid, id);
+        }
```

Comment:
else, setLoadInfo to empty, otherwise previous values are still displayed when you empty "id" field